cluster/etcd_service_discovery.go

```diff
@@ -644,14 +651,18 @@ func (sd *etcdServiceDiscovery) watchEtcdChanges() {
 			case wResp, ok := <-chn:
 				if wResp.Err() != nil {
 					logger.Log.Warnf("etcd watcher response error: %s", wResp.Err())
-					time.Sleep(100 * time.Millisecond)
+					time.Sleep(10 * time.Millisecond)
 				}
 				if !ok {
 					logger.Log.Error("etcd watcher died, retrying to watch in 1 second")
 					time.Sleep(1000 * time.Millisecond)
 					sd.InitETCDClient()
 					chn = sd.cli.Watch(context.Background(), "servers/", clientv3.WithPrefix())
 				}
+				// Wait for syncServers() to finish running to avoid conflicts
+				for syncServersRunning {
```

Comment:
instead of a for, it would be more goish and robust to use a channel or a waiting group

<- syncServersChan
or wg.Wait()

another option and probably more elegant/universal would be to use a mutex also syncServersMutex